packages/core/src/schema/inlineContent/types.ts

```diff
@@ -126,19 +126,27 @@ export type PartialInlineContent<
 > = PartialInlineContentElement<I, T>[] | string;
 
 export function isLinkInlineContent<T extends StyleSchema>(
-  content: InlineContent<any, T>
+  content: InlineContent<any, T> | undefined
```

Comment:
I want to be careful updating these types, what do these changes fix exactly / where was the error? shouldn't the typescript compiler break anyway if we had been passing undefined to the original functions?

Reply:
This was the sentry issue: https://blocknote-js.sentry.io/issues/32289262/?environment=vercel-production&query=is%3Aregressed&referrer=issue-stream&stream_index=2

I think you cannot assume that the index access into the array works, we should probably turn on `noUncheckedIndexAccess` on in our tsconfig: https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess

But, I stand by my previous statement that these functions are meant to assert that something is the correct type and therefore should also assert that something is there first before trying to assert that (i.e. check for undefined or null prior)

Reply:
Ok. I usually take the opposite approach; https://martinfowler.com/ieeeSoftware/failFast.pdf - let's discuss when you're back :)

Reply:
I'd agree with that in general, but I see a type-guard as a parser: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/ 😉

Reply:
Thanks for that and for the issue! Let's discuss when you're back!

Reply:
fixed the issue at different level so we can discuss when you're back, just good to get the fixes in now